tests/parametric/test_headers_tracecontext.py

```diff
@@ -807,6 +814,201 @@ def test_tracestate_w3c_p_inject(self, test_agent, test_library):
+        assert traceparent6.trace_id == "00000000000000000000000000000006"
+        assert "p:{:016x}".format(int(case6["span_id"])) in tracestate6["dd"]
+
+    # W3C Phase 3 to try adding the tag if the span id matches regardless of headers order(if tracecontext is accounted)
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library < "dotnet@2.51.0", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "cpp", reason="Not implemented")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @temporary_enable_optin_extract_first()
+    def test_tracestate_w3c_p_phase_3_extract_inject_extract_first(self, test_agent, test_library):
+        """
+        Ensure the last parent id tag is set according to the W3C phase 3 spec
+        """
+        with test_library:
+            # 1) Datadog and tracecontext headers, trace-id and span-id match
+            headers1 = make_single_request_and_get_inject_headers(
+                test_library,
+                [
+                    ["traceparent", "00-00000000000000000000000000000001-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
+                    ["x-datadog-trace-id", "1"],
+                    ["x-datadog-parent-id", "987654321"],
+                ],
+            )
+
+            # 2) Datadog headers only
+            headers2 = make_single_request_and_get_inject_headers(
+                test_library, [["x-datadog-trace-id", "2"], ["x-datadog-parent-id", "987654321"],],
+            )
+
+            # 3) Datadog and tracecontext headers, not x-datadog-trace-id, tid and the spans id match
+            headers3 = make_single_request_and_get_inject_headers(
+                test_library,
+                [
+                    ["traceparent", "00-00000000000000000000000000000003-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
+                    ["x-datadog-parent-id", "987654321"],
+                    ["x-datadog-tags", "_dd.p.tid=3"],
+                ],
+            )
+
+        traces = test_agent.wait_for_num_traces(3)
+
+        assert len(traces) == 3
+        case1, case2, case3 = traces[0][0], traces[1][0], traces[2][0]
+
+        # 1) Datadog and tracecontext headers, trace-id and span-id match
+        assert "_dd.parent_id" not in case1["meta"]
+
+        traceparent1, tracestate1 = get_tracecontext(headers1)
+        assert traceparent1.trace_id == "00000000000000000000000000000001"
+        assert "p:{:016x}".format(int(case1["span_id"])) in tracestate1["dd"]
```

Comment:
Yeah we don't need to validate the trace_ids, for asserting `p` in the  `tracestate1`  I thought it would be needed based on one of the points of the OKR:
> MUST inject the new p into the tracestate

But I assume this as been tested in another capacity already since the change for Phase 3 didn't require us changing our `inject` code.

Reply:
Addressed at: https://github.com/DataDog/system-tests/pull/2385/commits/e121ae9c14c25c73a1f91df597cc255d03e9f685